File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/AcademicYearService.java
package com.transkript.reportcard.business.service.i;

import com.transkript.reportcard.api.dto.AcademicYearDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.AcademicYear;
import java.util.List;

public interface AcademicYearService {
    
    EntityResponse create( AcademicYearDto var1);

    
    AcademicYearDto getDto( Long id);

    
    EntityResponse update( AcademicYearDto academicYearDto);

    List<AcademicYearDto> getAllDtos();

    void delete(Long var1);

    
    AcademicYear getEntity( Long var1);

    
    List<AcademicYear> getAllEntities();
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/AcademicYearService.kt
package com.transkript.reportcard.business.service.i

import com.transkript.reportcard.api.dto.AcademicYearDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.AcademicYear

interface AcademicYearService {
    fun create(academicYearDto: AcademicYearDto?): EntityResponse?
    val academicYears: List<AcademicYearDto?>?
    fun getDto(id: Long?): AcademicYearDto?
    fun update(id: Long?, academicYearDto: AcademicYearDto?): EntityResponse?
    fun delete(id: Long?)
    fun getEntity(academicYearId: Long?): AcademicYear?
    val academicYearEntities: List<AcademicYear?>?
}
